test/framework/management_cluster.go

```diff
@@ -125,3 +128,33 @@ func InitManagementCluster(ctx context.Context, input *InitManagementClusterInpu
 
 	return managementCluster
 }
+
+// WaitForDeploymentsAvailableInput is the input for WaitForDeploymentsAvailable.
+type WaitForDeploymentsAvailableInput struct {
+	Getter     Getter
+	Deployment *appsv1.Deployment
+}
+
+// WaitForDeploymentsAvailable waits until the Deployment has status.Available = True, that signals that
+// all the desired replicas are in place.
+// This can be used to check if Cluster API controllers installed in the management cluster are working.
+func WaitForDeploymentsAvailable(ctx context.Context, input WaitForDeploymentsAvailableInput, intervals ...interface{}) {
+	By(fmt.Sprintf("waiting for deployment %s/%s to be available", input.Deployment.GetNamespace(), input.Deployment.GetName()))
+	Eventually(func() bool {
+		deployment := &appsv1.Deployment{}
+		key := client.ObjectKey{
+			Namespace: input.Deployment.GetNamespace(),
+			Name:      input.Deployment.GetName(),
+		}
+		if err := input.Getter.Get(ctx, key, deployment); err != nil {
+			return false
+		}
+		for _, c := range deployment.Status.Conditions {
+			if c.Type == appsv1.DeploymentAvailable && c.Status == corev1.ConditionTrue {
+				return true
+			}
+		}
+		return false
+
+	}, intervals...).Should(BeTrue(), "Deployment %s/%s failed to get status.Available = True condition")
```

Comment:
Should this have been a `fmt.Sprintf("Deployment %s/%s failed to get status.Available = True condition", input.Deployment.GetNamespace(), input.Deployment.GetName())`

Reply:
@wfernandes good catch!
fixed by https://github.com/kubernetes-sigs/cluster-api/pull/2864